readme.txt
Usage:
  smurf from <from> to <to> [delay <delay>] 
  smurf -h | --help
  smurf --version
Options:
  -h --help     Show this screen.
  --version     Show version.



#this is the api

#install
#git clone https://....
#cd ...
#pipenv shell
#pip install -e .



for example try:
$ ./smurf  from "merhaba dünya:)" to "(:hello world" delay 0.1
 



history.txt
zo  2 jun 2019 14:06:28 CEST  first publish on github


smurf/hug_api.py
from itertools import cycle
import hug
import smurf
from smurf.docopt_cli import ri
#123456789012345678901234567890123456789012 #
class Smurf():                              #
 def __init__(i,n):                         # 
  i.n=n                                     #
  i.ml=[]                                   #
 def pcs(i,s,t=0.2):                        #
  i.ml.append(s)                            #
#123456789012345678901234567890123456789012 #
_examples="_f=hello&_t=world"               #
@hug.get('/smurf', examples=_examples)      #
def get_smurf(_f:hug.types.text,            #
              _t:hug.types.text):           #
 '''get a smurf, a morfing string'''        #
 sm=Smurf('smurf')                          #
 print("from=%s&to=%s"%(_f,_t))             #
 r=smurf.docopt_cli.m(_f,_t,0,C=sm.pcs)    #
 return sm.ml                               #
#123456789012345678901234567890123456789012 #
_examples="_f=hello&_t=world"               #
@hug.get('/circle', examples=_examples)     #
def get_circle(_f:hug.types.text,           #
               _t:hug.types.text):          #
 '''get a smurf, a morfing string'''        #
 sm1=Smurf('smurf1')                        #
 sm2=Smurf('smurf2')                        #
 print("from=%s&to=%s"%(_f,_t))             #
 r=smurf.docopt_cli.m(_f,_t,0,C=sm1.pcs)    #
 r=smurf.docopt_cli.m(_t,_f,0,C=sm2.pcs)    #
 return sm1.ml+sm2.ml                       #
#123456789012345678901234567890123456789012 #
@hug.get('/cycle', examples=_examples)      #
def get_cycle(_f:hug.types.text,            #
              _t:hug.types.text):           #
 '''get a smurf, a morfing string'''        #
 global ml                                  #
 ml=[]                                      #
 print("from=%s&to=%s"%(_f,_t))             #
 r=smurf.docopt_cli.m(_f,_t0,sm.pcs)        #
 r=smurf.docopt_cli.m(_t,_f,_d)             #
 #res=r+r[1:-1]                             #
 cres=cycle(ml)                             #
 for l in cres:                             #
  print('yielding',l)                       #
  yield l                                   #
  #BROKEN, fix this!drip?                   #
 #return ml                                 #
#123456789012345678901234567890123456789012 #
v={}                                        #
_examples="_i=42"                           #
@hug.get('/get', examples=_examples  )      #
def get_random(_i:hug.types.text):          #
 '''get a random string from a smurf'''     #
 global ml                                  #
 global v                                   #
 if _i not in v:                            #
  tg='merhaba dünya:)'                      #
  eg='hello world:)'                        #
  sm=Smurf('greet')                         #
  r=smurf.docopt_cli.m(tg,eg,0,C=sm.pcs)    #
  v[_i]=sm.ml                               #
 ml=v[_i]                                   #
 r=ri(1,len(ml)-1)                          #
 rs=ml[r]                                   #
 print("id=%s"%(_i))                        #
 print("r=%d"%(r))                          #
 print("rs=%s"%(rs))                        #
 return rs                                  #
#123456789012345678901234567890123456789012 #


smurf/docopt_cli.py
#!/usr/bin/env python3                       
#123456789012345678901234567890123456789012 #
import random                                 
import time                                  
import sys                                   
from docopt import docopt                    
#123456789012345678901234567890123456789012 #
def myclear():  #copy from click.clear      #
 '''Clears the terminal screen.  This will  #
 have the effect of clearing'''             #
 sys.stdout.write('\033[2J\033[1;1H')       #
#123456789012345678901234567890123456789012 #
_docopt="""smurf                            
 Usage:                                     
  smurf from <from> to <to> [delay <delay>] 
  smurf demo                                
  smurf -h | --help                         
  smurf --version                           
Options:                                    
  -h --help          Show this screen.      
  --version          Show version.          
  --debug            Show debug messages,   
                     for nerdy geeks        
  -b  --bare         only show morfing      
"""                                         
#123456789012345678901234567890123456789012 #
ri=random.randint                           #
def pcs(s,t=0.2):                           #
 myclear()                                  #
 print(s)                                   #
 time.sleep(t)                              #
T=0.7                                       #
#123456789012345678901234567890123456789012 #
_msg='''{m}'''                              #
_msg='''{f}{m}{t}'''                        #
_msg='''smurfing...                         #
from: '{from}'                              #
morf: '{morf}'                              #
  to: '{to}'                                #
'''                                         #
_msg='''{m}'''                              #
def ms(f,m,t):                              #
 d={}                                       #
 d['f']=d['from']=f                         # 
 d['m']=d['morf']=m                         #
 d['t']=d['to']=t                           #
 return _msg.format(**d)                    #
#123456789012345678901234567890123456789012 #
def m(f,t,d,C=pcs):                         #
 '''for morfing a string randomly from a    #
    source string to a random string        #
    for example,try:                        #
    set t 'merhaba:)'                       #
    set e '(:hello'                         #
    $ smurf from $t to $e d 0.3             #
              /todos:multiline              #
             /feedback welcome:)            #
            /colors >>>smurfing:blue        #
           /        -:red                   #
          /         +:green                 #
         /          ~:orange                #
        /            fade back              #
       /streams mqtt,kafka,strapi           #
      /within pipes, make safe share        #
     /use within hexy, simphex, cells       #
    /use within bu3 as hiding in foam       #
   /use for partly safe sharing a string    #
  /recognize of a random step from smurf    #
 /calculate edit distances/entropy diffing  #
/use perl style string a..z:abcdefghi...xyz #
 '''                                        #
#123456789012345678901234567890123456789012 #
 global T                                   #
 T=d                                        #
 if f==t:                                   #
  pcs('''sorry, refusing to smurf,          #
  there is no path to follow''')            #
  return                                    #
 msg=ms(f,f,t)                              #
 pcs(msg,T)                                 #
 of=f                                       #
 ot=t                                       #
 lf=len(f)                                  #
 lt=len(t)                                  #
 if lf>lt:                                  #
  while lf>lt:                              #
   r=ri(0,lf-1)                             #
   f=f[:r]+f[r+1:]                          #
   C(ms(of,f,t),T)                          #
   lf=len(f)                                #
 if lf<lt:                                  #
  while lf<lt:                              #
   r=ri(0,lf-1)                             #
   f=f[:r]+t[-r]+f[r:]                      #
   C(ms(of,f,t),T)                          #
   lf=len(f)                                #
 n=0                                        #
 l=[0 for i in range(lt)]                   #
 while n<lt:                                #
  r=ri(0,lt-1)                              #
  if l[r]:continue                          #
  n+=1                                      #
  l[r]=1                                    #
  c=t[r]                                    #
  #t=t[:r]+t[r+1:]                          #
  f=f[:r]+c+f[r+1:]                         #
  C(ms(of,f,t),T)                           #
  lt=len(t)                                 #
 C(ms(of,f,t),T)                            #
                                            #
                                            #
 #thnx='''\nthats all folks!                #
 #thanks for watching:)'''                  #
 #pcs(msg+f+thnx,T)                         #
                                            #
#123456789012345678901234567890123456789012 #
def go(n,d):                                #
 if n in d and d[n]:                        #
  return d['<'+n+'>']                       #
#123456789012345678901234567890123456789012 #
DEBUG=True                                  #
def debug(**k):                             #
 print(**k)                                 #
deb=debug                                   #
#123456789012345678901234567890123456789012 #
def main():                                 #
 ad = docopt(_docopt,                       #
             version='smurf 2019.05.19')    #
 #deb({'a':ad})                             #
 #deb(a=ad)                                 #
 f=go('from',ad)                            #
 t=go('to',ad)                              #
 bare=go('bare',ad)                         #
 d=go('delay',ad)                           #
 #o=go('options',ad)                        #
 if d:d=float(d)                            #
 if not d:d=T                               #
                                            #
 #t='merhaba dünya:)'                       #
 #e='hello world:)'                         #
 #if demo:i.m(f=t,t=e,d=0.2)                #
 m(f=f,t=t,d=d)                             #
#123456789012345678901234567890123456789012 #
if __name__ == '__main__':                  #
 main()                                     #
#123456789012345678901234567890123456789012 #
